### PR TITLE
Add highlighting for standard links and dropdown

### LINKED ISSUE #1390: Extend navbar highlighting to <a> tags and dropdowns
**Is your request related to a problem?**
This issue is a followup of #1379. Currently, the top-level links of navbars are being highlighted when there's a match found between `window.location.href` and in its children anchortags `a.href` (with highlighting rules based on the `highlight-on` and `default-highlight-on` attribute). It would be good to extend this highlighting to *individual* links/anchortags, as well as *individual* dropdown items (navbars support both dropdowns and standard links).

Related code snippet:
```javascript
if (this.isSibling(url, a.href) || this.isChild(url, a.href)) {
    li.classList.add('current');
    return;
}
```

A referenced comment from the discussion:
```
Just a thought, is it possible to:

- apply the current styling on <a> tags
- additionally apply it to only dropdowns?

Reason being we may want to highlight both the parent (dropdown header) and child (dropdown item) links
```

**Describe the solution you'd like**

Changes in styling in `markbind.css`, as well as changes in logic in `Navbar.vue` under `highlightLink(url)`. This changes will most likely only involve changes on the front-end. Highlighting should ideally be subtle (e.g lighter text for dark navbars, darker text for light navbars)

**Describe alternatives you've considered**

No alternatives have been considered yet

**Additional context**

More about navbar highlighting: https://markbind.org/userGuide/usingComponents.html#navbars

### PR DESCRIPTION
**Pull request by jonahtanjz**
**What is the purpose of this pull request?**

- [ ] Documentation update
- [ ] Bug fix
- [x] Feature addition or enhancement
- [ ] Code maintenance
- [ ] Others, please explain:

**Overview of changes:**
Fixes #1390. This adds highlighting to individual links/anchortags, as well as individual dropdown items. For anchortags, if `window.location.href` matches its `a.href`, it will be assigned the `current` class. For dropdowns, the parent (dropdown header) will be assigned the `current` class and the child (dropdown item) will be assigned the `dropdown-current` class which will highlight both the parent and child links.

**Anything you'd like to highlight / discuss:**
Implementation and code quality

**Testing instructions:**
1. Create a navbar with dropdown and standard links
1. Click on each link
1. The link that the page is currently on will be highlighted

**Proposed commit message: (wrap lines at 72 characters)**
Extend navbar highlighting to standard links and dropdowns

---

**Checklist:** :ballot_box_with_check:

- [x] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No blatantly unrelated changes
- [x] Pinged someone for a review!

### CODE REVIEW

File: packages/vue-components/src/Navbar.vue [hunk 1]
```diff
@@ -132,9 +132,10 @@
         const navLis = Array.from(this.$el.querySelector('.navbar-nav').children);
         // attempt an exact match first
         for (const li of navLis) {
+          const standardLinks = [li];
```
Comment (ang-zeyu) [inline packages/vue-components/src/Navbar.vue:135]: hmm what's this for? 😮 

tried removing this and it worked fine still

File: packages/vue-components/src/Navbar.vue [hunk 2]
```diff
@@ -160,16 +165,25 @@
             if (hlMode === 'sibling-or-child') {
               if (this.isSibling(url, a.href) || this.isChild(url, a.href)) {
                 li.classList.add('current');
+                if (dropdownLinks.includes(a)) {
```
Comment (ang-zeyu) [inline packages/vue-components/src/Navbar.vue:168]: let's move L167-170 to a inner function and use that - less repitition

File: packages/core-web/src/styles/markbind.css [hunk 3]
```diff
@@ -463,3 +463,8 @@ button.copy-btn {
 .navbar.navbar-dark.bg-primary .navbar-nav .current {
     background: rgba(255, 255, 255, 0.05);
 }
+
+.dropdown-current {
```
Comment (ang-zeyu) [inline packages/core-web/src/styles/markbind.css:467]: you could use vue loader's deep selectors here to encapsulate it in navbar.vue

https://vue-loader.vuejs.org/guide/scoped-css.html

**Review by ang-zeyu — COMMENTED** [review]
Thanks! @jonahtanjz  Really clean implementation

Some comments, and one open question:

File: packages/vue-components/src/Navbar.vue (hunk 1 above)
Comment (jonahtanjz) [reply packages/vue-components/src/Navbar.vue:140]: Oh this is for cases where the navbar does not use the `li` tags. 
For example:
```
<header>
  <navbar type="dark">
    <a slot="brand" href="{{baseUrl}}/index.html" title="Home" class="navbar-brand">Your Logo</a>
    <a href="{{baseUrl}}/contents/topic1.html" class="nav-link">Topic 1</a>
    <a href="{{baseUrl}}/contents/topic2.html" class="nav-link">Topic 2</a>
    <a href="{{baseUrl}}/contents/topic3a.html" class="nav-link">Topic 3a</a>
    <a href="{{baseUrl}}/contents/topic3b.html" class="nav-link">Topic 3b</a>
  </navbar>
</header>
```
In this case, without this line, the highlighting would not work properly as `navLinks` only looks for anchortags within `li`.

File: packages/vue-components/src/Navbar.vue (hunk 2 above)
Comment (jonahtanjz) [reply packages/vue-components/src/Navbar.vue]: Alright thanks. I have shifted this to an inner function.

File: packages/core-web/src/styles/markbind.css (hunk 3 above)
Comment (jonahtanjz) [reply packages/core-web/src/styles/markbind.css]: I have moved this block into navbar,vue. Thanks.

**Comment by jonahtanjz** [comment]
Thanks for the review @ang-zeyu. One question, I've also written additional test site but when I ran `npm run updatetest`, alot of other test files got modified in the process. Not sure if this is intended or I'm not doing it correctly.

**Comment by ang-zeyu** [comment]
> Thanks for the review @ang-zeyu. One question, I've also written additional test site but when I ran `npm run updatetest`, alot of other test files got modified in the process. Not sure if this is intended or I'm not doing it correctly.

This (ideally) needs vue snapshot tests (which tests post-vue-render diffs), not our custom functional tests which we *mostly* use to detect unwanted changes on the backend (automatically, without serving them up)


Vue snapshot tests were added rather recently in https://github.com/MarkBind/vue-strap/pull/146 though, and only for `<box>`es. We don't have a whole lot of them yet.

I'm ok with proceeding without tests for now, we could add them for navbars (and etc.) in a later PR together. 🙂

File: packages/vue-components/src/Navbar.vue (hunk 1 above)
Comment (ang-zeyu) [reply packages/vue-components/src/Navbar.vue:140]: thanks for the explanation; nice catch 👍

**Review by ang-zeyu — APPROVED** [review]
Lgtm, are you intending to make anymore changes? (seeing that this is a draft)

**Comment by jonahtanjz** [comment]
Thanks @ang-zeyu for the clarification. Nope I don't intend to make anymore changes and I have opened the PR now.

**Pull request merged**